src/object.d

```diff
@@ -1751,6 +1751,59 @@ unittest
 }
 
 
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
+ *
+ * Example:
+ * ---
+ * class MeaCulpa: Exception
+ * {
+ *     mixin StandardExceptionConstructor;
+ * }
+ * assertThrown!MeaCulpa(enforce!MeaCulpa(false));
+ *
+ * try throw new MeaCulpa;
+ * catch (MeaCulpa e)
+ * {
+ *     assert(e.file == __FILE__);
+ *     assert(e.line == __LINE__ - 4);
+ * }
+ * ---
+ *
+ * Note that if you want to pass in special arguments to your exception, you
+ * obviously can't use this mixin and have to write the constructor manually.
+ */
+mixin template StandardExceptionConstructor()
+{
+    this(string msg, string file = __FILE__, size_t line = __LINE__)
+    {
+        super(msg, file, line);
+    }
+}
+
+unittest
```

Comment:
ddoc-ed `unittest` blocks are put into the preceding ddoc comment as an example so that the example is tested while only having to be typed once. Code in Phobos that explicitly has the example in a function's documentation and then has a non-ddoc-ed `unittest` block to test it pre-dates the ddoc `unittest` block feature.